### ISSUE TITLE
indentation misaligned due to `where` in comment

### ISSUE THREAD

**Issue by nikomatsakis** [rust-lang/rust-mode#151]
This example indents as shown:

```rust
/// - there must not exist an edge U->V in the graph where:
#[derive(Clone, PartialEq, Eq)]
    pub struct Region {
    entry: BasicBlockIndex,
    leaves: BTreeMap<BasicBlockIndex, usize>,
}
```

But I would expect:

```rust
/// - there must not exist an edge U->V in the graph where:
#[derive(Clone, PartialEq, Eq)]
pub struct Region { // <-- this should be flush with left margin!
    entry: BasicBlockIndex,
    leaves: BTreeMap<BasicBlockIndex, usize>,
}
```

Presumably the angle-bracket matching code is going a bit awry here.

cc @MicahChalmer

**Comment by nikomatsakis** [comment]
Interestingly, removing the `#[derive]` makes things work.

**Comment by MicahChalmer** [comment]
It's got nothing to do with the `>` this time.  You can try removing it--the problem doesn't go away.  The trigger is actually the word "where" in the comment.  The code that looks for where clauses isn't checking that the word isn't part of a string or comment.

**Comment by MicahChalmer** [comment]
I changed the title of the issue to reflect the actual trigger...(title said it was due to the `>` but it's actually due to the `where`)

**Comment by mrBliss** [comment]
I'll fix this, I think I introduced this in b76e803768bd572424b3f111aa55269f5d8226fc.

**Comment by nikomatsakis** [comment]
thanks! 👍
